### ISSUE TITLE
NullPointerException while setting Annotation in corenlp 3.9.2

### ISSUE THREAD

**Issue by amitru** [stanfordnlp/CoreNLP#951]
I am getting below error, i am using pom.xml with below two dependencies and getting the same error, i am in java 1.8. The error is coming at line : -pipeline.annotate(document)

Code: - 

```
String result="";
        Properties props = new Properties();
        props.put("ner.applyFineGrained", "0");
        props.setProperty("annotators", "tokenize, ssplit, pos, lemma, ner, parse, dcoref,sentiment");
        StanfordCoreNLP pipeline = new StanfordCoreNLP(props);
        // create an empty Annotation just with the given text
        Annotation document = new Annotation(text);
        // run all Annotators on this text
        **pipeline.annotate(document);**
```

ERROR: 
```
Exception in thread "main" java.lang.NullPointerException
at edu.stanford.nlp.pipeline.DeterministicCorefAnnotator.setNamedEntityTagGranularity(DeterministicCorefAnnotator.java:93)
at edu.stanford.nlp.pipeline.DeterministicCorefAnnotator.annotate(DeterministicCorefAnnotator.java:173)
at edu.stanford.nlp.pipeline.AnnotationPipeline.annotate(AnnotationPipeline.java:76)
at edu.stanford.nlp.pipeline.StanfordCoreNLP.annotate(StanfordCoreNLP.java:637)
at 
```

Maven:

```
<dependency>
		    <groupId>edu.stanford.nlp</groupId>
		    <artifactId>stanford-corenlp</artifactId>
		    <version>3.9.2</version>
		</dependency>
		
		<dependency>
		    <groupId>edu.stanford.nlp</groupId>
		    <artifactId>stanford-corenlp</artifactId>
		    <version>3.9.2</version>
		    <classifier>models</classifier>
		</dependency>
```

**Comment by amitru** [comment]
Any updates?

**Comment by J38** [comment]
I'll have to fix this bug in Stanford CoreNLP 4.0.0, which is actually coming out fairly soon hopefully.  It doesn't look like coref will work if you shut off the fine-grained NER, so it causes an NPE.

**Comment by amitru** [comment]
Thanks for the update, but i have not done any config changes i've just taken the project and running it, if you can suggest any workaround for now will be helpful...

**Comment by AngledLuffa** [comment]
Do you have an example of some text that causes this crash?

**Comment by AngledLuffa** [comment]
This is now fixed in version 4.0.0, I believe.
